integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala

```diff
@@ -417,59 +396,30 @@ class CarbonFileMetastore extends CarbonMetaStore {
     thriftWriter.open(FileWriteOperation.OVERWRITE)
     thriftWriter.write(thriftTableInfo)
     thriftWriter.close()
-    updateSchemasUpdatedTime(touchSchemaFileSystemTime())
+    updateSchemasUpdatedTime(thriftTableInfo.getFact_table.getTable_id, thriftTableInfo
+      .getFact_table.getSchema_evolution.schema_evolution_history.get(0).time_stamp)
     identifier.getTablePath
   }
 
-  protected def addTableCache(
+  protected def addCarbonTableToCache(
       tableInfo: table.TableInfo,
-      absoluteTableIdentifier: AbsoluteTableIdentifier): ArrayBuffer[CarbonTable] = {
+      absoluteTableIdentifier: AbsoluteTableIdentifier): Unit = {
     val identifier = absoluteTableIdentifier.getCarbonTableIdentifier
     CarbonMetadata.getInstance.removeTable(tableInfo.getTableUniqueName)
     removeTableFromMetadata(identifier.getDatabaseName, identifier.getTableName)
     CarbonMetadata.getInstance().loadTableMetadata(tableInfo)
-    metadata.writeLock.lock()
-    metadata.carbonTables +=
-      CarbonMetadata.getInstance().getCarbonTable(identifier.getTableUniqueName)
-    metadata.writeLock.unlock()
-    metadata.carbonTables
   }
 
   /**
-   * This method will remove the table meta from catalog metadata array
-   *
-   * @param dbName
-   * @param tableName
+   * This method will remove the table meta from CarbonMetadata cache.
    */
   def removeTableFromMetadata(dbName: String, tableName: String): Unit = {
-    val carbonTableToBeRemoved: Option[CarbonTable] = getTableFromMetadataCache(dbName, tableName)
-    carbonTableToBeRemoved match {
-      case Some(carbonTable) =>
-        metadata.writeLock.lock()
-        metadata.carbonTables -= carbonTable
-        metadata.writeLock.unlock()
-      case None =>
-        if (LOGGER.isDebugEnabled) {
-          LOGGER.debug(s"No entry for table $tableName in database $dbName")
-        }
-    }
     CarbonMetadata.getInstance.removeTable(dbName, tableName)
   }
 
   private def updateMetadataByWrapperTable(
```

Comment:
This method looks redundant, we can call` addCarbonTableToCache` instead of it.

Reply:
done